src/main/java/org/eln2/mc/common/content/Wire.kt

```diff
@@ -0,0 +1,315 @@
+import org.eln2.mc.Eln2
+import org.eln2.mc.client.render.MultipartBlockEntityInstance
+import org.eln2.mc.client.render.PartialModels
+import org.eln2.mc.common.cells.CellRegistry
+import org.eln2.mc.common.cells.foundation.CellBase
+import org.eln2.mc.common.cells.foundation.CellPos
+import org.eln2.mc.common.cells.foundation.objects.ElectricalComponentInfo
+import org.eln2.mc.common.cells.foundation.objects.ElectricalObject
+import org.eln2.mc.common.cells.foundation.objects.ResistorBundle
+import org.eln2.mc.common.cells.foundation.objects.SimulationObjectSet
+import org.eln2.mc.common.cells.foundation.Conventions
+import org.eln2.mc.common.parts.foundation.CellPart
+import org.eln2.mc.common.parts.foundation.ConnectionMode
+import org.eln2.mc.common.parts.foundation.IPartRenderer
+import org.eln2.mc.common.parts.foundation.PartPlacementContext
+import org.eln2.mc.common.space.DirectionMask
+import org.eln2.mc.common.space.RelativeRotationDirection
+import org.eln2.mc.extensions.ModelDataExtensions.blockCenter
+import org.eln2.mc.extensions.ModelDataExtensions.zeroCenter
+import org.eln2.mc.extensions.NbtExtensions.getRelativeDirection
+import org.eln2.mc.extensions.NbtExtensions.putRelativeDirection
+import org.eln2.mc.extensions.QuaternionExtensions.times
+import org.eln2.mc.extensions.Vec3Extensions.times
+import org.eln2.mc.extensions.Vec3Extensions.toVec3
+import org.eln2.mc.integration.waila.IWailaProvider
+import org.eln2.mc.integration.waila.TooltipBuilder
+import java.util.concurrent.atomic.AtomicReference
+import kotlin.math.abs
+
+// This organisation approach is inspired by a suggestion from jrddunbr.
+// Having the content in one file does help navigate more easily.
+
+/**
+ * The Wire Object has a single resistor bundle. The Internal Pins of the bundle are connected to each other, and
+ * the External Pins are exported to other Electrical Objects.
+ * */
+class WireObject : ElectricalObject(), IWailaProvider {
+    private val resistors = ResistorBundle(0.05)
+
+    /**
+     * Gets or sets the resistance of the bundle.
+     * Only applied when the circuit is re-built.
+     * */
+    var resistance : Double = resistors.resistance
+        set(value) {
+            field = value
+            resistors.resistance = value
+        }
+
+    override fun offerComponent(neighbour: ElectricalObject): ElectricalComponentInfo {
+        return resistors.getOfferedResistor(directionOf(neighbour))
+    }
+
+    override fun recreateComponents() {
+        resistors.clear()
+    }
+
+    override fun registerComponents(circuit: Circuit) {
+        resistors.register(connections, circuit)
+    }
+
+    override fun build() {
+        resistors.connect(connections, this)
+
+        // Is there a better way to do this?
+
+        // The Wire uses a bundle of 4 resistors. Every resistor's "Internal Pin" is connected to every
+        // other resistor's internal pin. "External Pins" are offered to connection candidates.
```

Comment:
In Eln1, it would make the resistors as the connections were made. This had the plus of not increasing matrix size and allowing line optimization since there were no external pins that were floating from a disconnected resistor.